alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/properties/Cognitive2D.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2010-2022, Danilo Pianini and contributors
+ * listed, for each module, in the respective subproject's build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.model.implementations.properties
+
+import it.unibo.alchemist.model.cognitiveagents.CognitiveModel
+import it.unibo.alchemist.model.interfaces.Molecule
+import it.unibo.alchemist.model.interfaces.Node
+import it.unibo.alchemist.model.interfaces.properties.CognitiveProperty
+import it.unibo.alchemist.model.interfaces.environments.Physics2DEnvironment
+
+/**
+ * The node's [CognitiveModel].
+ */
+class Cognitive2D<T> @JvmOverloads constructor(
+    /**
+     * The environment in which the node moves.
+     */
+    environment: Physics2DEnvironment<T>,
+    node: Node<T>,
+    /**
+     * The molecule associated with danger in the environment.
+     */
+    danger: Molecule? = null,
+) : CognitiveProperty<T> by Cognitive(
+    environment,
+    node,
+    danger,
+)
```

Comment:
This class merely calls a super constructor, do we need this at all?

Reply:
I maintained the design approach by leaving a general class and extending it for the 2D environment. Should i make it a typealias?